lib/auth/auth_with_roles.go

```diff
@@ -3160,6 +3157,55 @@ func (a *ServerWithRoles) DeleteAllAppSessions(ctx context.Context) error {
 	return nil
 }
 
+// DeleteUserAppSessions delete all user’s application sessions.
+func (a *ServerWithRoles) DeleteUserAppSessions(ctx context.Context, req *proto.DeleteUserAppSessionsRequest) error {
+	// First, check if the current user can delete the request user sessions.
+	if err := a.checkPermissionToDeleteUserWebSession(req.Username); err != nil {
+		return trace.Wrap(err)
+	}
+
+	sessions, err := a.authServer.GetAppSessions(ctx)
```

Comment:
Can we either make a backend method that returns user's app sessions instead of checking `req.Username != session.GetUser()` below, or encapsulate it in the helper method? Something like `GetAppSessionsFor(ctx, <user>)`.